### PR TITLE
DPTP-4138: Add additional tests for node_Architecture

### PR DESCRIPTION
**Pull request by deepsm007**
/cc @droslean

### CODE REVIEW

File: pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml [hunk 1]
```diff
@@ -684,3 +684,337 @@
...
+      image: registry.ci.openshift.org/ci/entrypoint-wrapper:latest
+      name: cp-entrypoint-wrapper
+      resources: {}
+      terminationMessagePolicy: FallbackToLogsOnError
+      volumeMounts:
+      - mountPath: /tmp/entrypoint-wrapper
+        name: entrypoint-wrapper
+    nodeName: node-name
+    nodeSelector:
+      kubernetes.io/arch: amd64
```
Comment (droslean) [inline pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml:831]: This should be `arm64`, right?

Comment (deepsm007) [reply pkg/steps/multi_stage/testdata/zz_fixture_TestGeneratePods.yaml:831]: Yes, these were auto generated

File: pkg/steps/multi_stage/gen.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ type generatePodOptions struct {
 func defaultGeneratePodOptions() *generatePodOptions {
 	return &generatePodOptions{
 		IsObserver:       false,
-		NodeArchitecture: string(api.NodeArchitectureAMD64),
+		NodeArchitecture: "",
```
Comment (droslean) [inline pkg/steps/multi_stage/gen.go:60]: But we want to default it to amd64. Is there any particular reason of removing it?

Comment (deepsm007) [reply pkg/steps/multi_stage/gen.go:60]: I missed adding back after testing, nice catch

File: pkg/steps/multi_stage/gen.go [hunk 3]
```diff
@@ -165,6 +165,9 @@ func (s *multiStageTestStep) generatePods(
 				pod.Spec.DNSPolicy = coreapi.DNSNone
 			}
 		}
+		if step.NodeArchitecture != nil {
+			pod.Spec.NodeSelector["kubernetes.io/arch"] = string(*step.NodeArchitecture)
```
Comment (droslean) [inline pkg/steps/multi_stage/gen.go:169]:
```suggestion
			pod.Spec.NodeSelector[corev1.LabelArchStable] = string(*step.NodeArchitecture)
```

**Review by droslean — APPROVED** [review]

**Pull request merged**